packages/members-ssr/index.js

```diff
@@ -203,6 +215,18 @@ class MembersSSR {
         const token = Array.isArray(query.token) ? query.token[0] : query.token;
         const member = await this._getMemberDataFromToken(token);
 
+        // perform and store geoip lookup for members when they log in
+        // TODO: is it preferable to do this here or in members-ssr consumer?
+        if (!member.geolocation) {
+            try {
+                await this._setMemberGeolocationFromIp(member.email, req.ip);
+            } catch (err) {
+                // no-op, we don't want to stop anything working due to
+                // geolocation lookup failing
+                // TODO: would be useful to log the failure here
```

Comment:
I couldn't see any logging in this library. Is this intentional?

Reply:
Yes this is, the idea was that the consumer of the library would do the logging.

Whether or not this is a good pattern, I'm not sure :thinking:

I guess this ties into the question in `members-api` about whether we should be throwing errors there or not.

I'm wondering if _that_ is where the logging should occur?

I'm also wondering if this shouldn't be a fire n forget like
`this._setMemberGeolocationFromIp(member.email, req.ip).catch(noop);`

Reply:
> I guess this ties into the question in members-api about whether we should be throwing errors there or not. I'm wondering if that is where the logging should occur?

The problem with moving the logging without no throw into `members-api` is that it's one further step away from the application-level consumer.

In `members-api`, if we're only logging and not throwing an error then we're stopping any consumers of that library from controlling the flow, maybe one consumer wants to blow up and log whereas another wants everything to happen silently and doesn't care if it fails 🤔

Unless I'm missing something, if we're not passing logging config down through each layer (`Ghost`->`members-ssr`->`members-api`) then the library-level logs will not necessarily reach where we expect?

Reply:
> The problem with moving the logging without throwing into members-api is that it's one further step away from the application-level consumer.

So from how I see it, it moves the logging one step _closer_ to the application :grin: 

Although `members-api` is like this library - it _is_ a core part of Ghost - and the original design of it being a library isn't really valid any more, I would consider the `members-api` module just as much a part of ghost as the `core/frontend` directory in Ghost - if that makes sense?

So I see the "structure" more like

`Ghost -> members-api`

`Frontend -> members-ssr -> Ghost (-> members-api)`

Reply:
Resorted to using `debug()` so that errors can be traced when investigating but keeps in-line with no logging in this module - I guess if we want to add logging in the future we'd need to pass in a logger instance as we do with `members-api`